### PR TITLE
Disable FP64 emulation with tests

### PR DESCRIPTION
**Pull request by Alexandr-Solovev**
# Description
Reopen https://github.com/oneapi-src/oneDAL/pull/2252 with some additions

### CODE REVIEW

File: cpp/oneapi/dal/backend/primitives/selection/test/row_partitioning_dpc.cpp [hunk 1]
```diff
@@ -133,6 +133,7 @@ TEMPLATE_LIST_TEST_M(row_partitioning_test,
                      "row partitioning test on single random row",
                      "[row_partitioning][small]",
                      partitioning_types) {
+    SKIP_IF(this->not_float64_friendly());
     using float_t = TestType;
     std::int64_t rows = 1;
     std::int64_t cols = 17;
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/backend/primitives/selection/test/row_partitioning_dpc.cpp:139]:
```suggestion
    constexpr std::int64_t rows = 1;
    constexpr std::int64_t cols = 17;
```

File: cpp/oneapi/dal/backend/primitives/selection/test/row_partitioning_dpc.cpp [hunk 2]
```diff
@@ -149,6 +150,7 @@ TEMPLATE_LIST_TEST_M(row_partitioning_test,
                      "row partitioning test (two rows)",
                      "[row_partitioning][small]",
                      partitioning_types) {
+    SKIP_IF(this->not_float64_friendly());
     using float_t = TestType;
     std::int64_t rows = 2;
     std::int64_t cols = 17;
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/backend/primitives/selection/test/row_partitioning_dpc.cpp:156]:
```suggestion
    constexpr std::int64_t rows = 2;
    constexpr std::int64_t cols = 17;
```

File: cpp/oneapi/dal/backend/primitives/selection/test/row_partitioning_dpc.cpp [hunk 3]
```diff
@@ -165,6 +167,7 @@ TEMPLATE_LIST_TEST_M(row_partitioning_test,
                      "row partitioning test (unaligned block)",
                      "[row_partitioning][small]",
                      partitioning_types) {
+    SKIP_IF(this->not_float64_friendly());
     using float_t = TestType;
     std::int64_t rows = 17;
     std::int64_t cols = 37;
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/backend/primitives/selection/test/row_partitioning_dpc.cpp:173]:
```suggestion
    constexpr std::int64_t rows = 17;
    constexpr std::int64_t cols = 37;
```

File: cpp/oneapi/dal/backend/primitives/selection/test/row_partitioning_dpc.cpp [hunk 4]
```diff
@@ -199,6 +203,7 @@ TEMPLATE_LIST_TEST_M(row_partitioning_test,
                      "row partitioning test (end of single row)",
                      "[row_partitioning][small]",
                      partitioning_types) {
+    SKIP_IF(this->not_float64_friendly());
     using float_t = TestType;
     std::int64_t rows = 1;
     std::int64_t cols = 35;
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/backend/primitives/selection/test/row_partitioning_dpc.cpp:209]:
```suggestion
    constexpr std::int64_t rows = 1;
    constexpr std::int64_t cols = 35;
```

File: cpp/oneapi/dal/backend/primitives/selection/test/row_partitioning_dpc.cpp [hunk 5]
```diff
@@ -216,6 +221,7 @@ TEMPLATE_LIST_TEST_M(row_partitioning_test,
                      "row partitioning test (partial unaligned block)",
                      "[row_partitioning][small]",
                      partitioning_types) {
+    SKIP_IF(this->not_float64_friendly());
     using float_t = TestType;
     std::int64_t rows = 17;
     std::int64_t cols = 37;
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/backend/primitives/selection/test/row_partitioning_dpc.cpp:227]:
```suggestion
    constexpr std::int64_t rows = 17;
    constexpr std::int64_t cols = 37;
```

**Comment by ethanglaser** [comment]
Should CI be checked with infra #605?

**Comment by napetrov** [comment]
i think they can be independently integrated - as long as this one is green we can integrate it and check with infra change

**Review by KulikovNikita — APPROVED** [review]

**Comment by KulikovNikita** [comment]
I would also suggest to add a compile time check for the `float_algo_fixture` template argument. It should always be a floating point type

**Review by ethanglaser — APPROVED** [review]

**Comment by napetrov** [comment]
> I would also suggest to add a compile time check for the `float_algo_fixture` template argument. It should always be a floating point type

on this one i suggest to discuss offline and consider adding in next PR if needed

**Pull request merged**